### DISCUSSION TITLE
Emit Event

### DISCUSSION THREAD

**Discussion by MentalGear** [directus/directus#16605] [Archive]
Currently there is [no explanation](https://docs.directus.io/configuration/flows.html) on  "Emit Event" in Flows. I'd be great to add one to the guide or better as a tooltip in the UI.

**Comment by azrikahar** [comment]
If the "Emit Event" here means the "Emit Events" option in Flow's create/read/update/delete data operations:

![](https://user-images.githubusercontent.com/42867097/203908174-bd8f1de3-c50f-4996-a3f2-aaa995451519.png)

There's should be an detailed explanation of it in each of the operation! More specifically, the warning block near the end of each of them: 

![](https://user-images.githubusercontent.com/42867097/203908362-1c70e8ce-571b-46ad-8ec3-7587cbc47666.png)

Direct link to above docs: https://docs.directus.io/configuration/flows/operations.html#create-data

Direct quote for readability (as of the posting of this comment in case it's updated in the future):

> Emit Events toggles the event's _visibility_ throughout Directus. For example, if togged on, this operation will trigger relevant event hooks in other flows or custom extensions. If toggled off, the operation will not trigger other event hooks. Imagine a situation where you have a flow being triggered by `<collection>.items.create` which contains an operation that then tries to create another item in that `<collection>`. This would throw an infinite loop where the operation triggers its own flow, endlessly. However, if you toggle Emit Events off, then this operation no longer triggers other event hooks.

**Reply by MentalGear** [reply]
Thanks for the detailed explications!

Having to put the same big warning 4x in the docs, is a good indicator that it is not clear enough. Alone the name "Emit Events", first thoughts that come are: "Does this need to be on for other flow steps to receive the data/event from this one?", "Is this an extra option ( new submenu  opens ) to emit a custom event along with this?". Also the context in which it is triggered, e.g. for flows or  the whole of "directus", is not clear.


### Clarification Proposal
I'd propose to rename __"Emit Events"__ to something similar to __"Should trigger Event Hooks across Directus"__, and maybe also add the doc's WARNING Explication directly under the toggle as a expandable div for context ( there's enough room below anyhow ).

**Reply by azrikahar** [reply]
> Having to put the same big warning 4x in the docs, is a good indicator that it is not clear enough. 

Granted I believe another factor with this is there has been a number of questions where they are technically answered in another section (or operations in this context) but it may not be seen when a user is only looking at one section. But I'm sure @erondpowell our docs overseer would have a deeper insight to this and can offer more thoughts on the potential clarifications/changes 😄

> I'd propose to rename "Emit Events" to something similar to "Should trigger Event Hooks across Directus"

As for renaming the option in the App, that's an interesting suggestion!

Perhaps another alternative would be to include a note for this field, but would be tricky to describe it without being too verbose in the App drawer (of course the docs can be as verbose as needed). @benhaynes any thoughts on this?

**Reply by benhaynes** [reply]
I agree, we wouldn't change the field name (`Emit Events`), but we can add a note (see below). Let's have @rijkvanzanten confirm that I'm understanding this correctly.

```
These events will trigger other cascading hooks. Please be careful about accidentally creating infinite loops within self-referencing flows.
```

**Reply by MentalGear** [reply]
A note is already a big improvement, but for newcomers like me it could be more explicit:

```
If activated, this operation's events will trigger event listeners (hooks) directus-wide (including in other flows). For more details please consider the [docs](https://docs.directus.io/configuration/flows/operations.html).
Please be careful about accidentally creating infinite loops within (self or cross-referencing) flows.
```

Eventually, and I know this is probably significantly harder dircetus vs only current flow-wide, it'd be great if Directus could check and prevent infinite loops by showing an alert.
